### DISCUSSION TITLE
About the future

### DISCUSSION THREAD

**Discussion by jihaohaaaa** [denoland/deno#12657] [General]
An idea has been echoed in my mind for many years, that is, to release a language with the same syntax, but with script (dynamic) version and compiled (static) version. Until I met JavaScript and typescript, I thought such a language was finally born. But I found that typescript still needs to be compiled into JavaScript, so that there is no real typescript runtime

Maybe a new language will be born at some point in the future. It has two aspects: script as a glue language to bond the advantages of various languages (the speed of C / C + + and a wide variety of libraries of JavaScript), and static compilation as a non interpretative language with excellent performance (compiled into machine code or relying on a visual machine like JVM).

I hope it will be deno.

**Comment by abflow** [comment]
Might it be AssemblyScript ?
https://www.assemblyscript.org/
